src/main/java/de/hhu/stups/plues/ui/controller/Reports.java

```diff
@@ -185,67 +270,242 @@ private void displayModuleAbstractUnitUnitSemesterConflicts(final Store store,
       }
     });
     moduleAbstractUnitUnitSemesterConflicts.setData(conflicts);
+
+    return conflicts;
   }
 
-  private void displayImpossibleCourseModuleAbstractUnitPairs(final Store store,
-      final ReportData reportData) {
-    impossibleCourseModuleAbstractUnitPairs.setData(
+  private Map<Course, Map<Module, Set<Pair<AbstractUnit>>>>
+      displayImpossibleCourseModuleAbstractUnitPairs(final Store store,
+                                                     final ReportData reportData) {
+    Map<Course, Map<Module, Set<Pair<AbstractUnit>>>> impossibleCourseModuleAbstractUnitPairs =
         reportData.getImpossibleCourseModuleAbstractUnitPairs()
         .entrySet().stream().collect(Collectors.toMap(
           entry -> store.getCourseByKey(entry.getKey()),
           entry -> entry.getValue().entrySet().stream().collect(Collectors.toMap(
             innerEntry -> store.getModuleById(innerEntry.getKey()),
             innerEntry -> innerEntry.getValue().stream().map(
-                pair -> new Pair<>(store.getAbstractUnitById(pair.getFirst()),
-                  store.getAbstractUnitById(pair.getSecond()))).collect(Collectors.toSet()))))));
+              pair -> new Pair<>(store.getAbstractUnitById(pair.getFirst()),
+              store.getAbstractUnitById(pair.getSecond()))).collect(Collectors.toSet())))));
+    this.impossibleCourseModuleAbstractUnitPairs.setData(
+        impossibleCourseModuleAbstractUnitPairs);
+
+    return impossibleCourseModuleAbstractUnitPairs;
   }
 
-  private void displayImpossibleCourseModuleAbstractUnits(final Store store,
-      final ReportData reportData) {
-    impossibleCourseModuleAbstractUnits.setData(reportData.getImpossibleCourseModuleAbstractUnits()
+  private Map<Course, Map<Module, Set<AbstractUnit>>>
+      displayImpossibleCourseModuleAbstractUnits(final Store store, final ReportData reportData) {
+    Map<Course, Map<Module, Set<AbstractUnit>>> impossibleCourseModuleAbstractUnits =
+        reportData.getImpossibleCourseModuleAbstractUnits()
         .entrySet().stream().collect(Collectors.toMap(
           entry -> store.getCourseByKey(entry.getKey()),
           entry -> entry.getValue().entrySet().stream().collect(Collectors.toMap(
             innerEntry -> store.getModuleById(innerEntry.getKey()),
             innerEntry -> innerEntry.getValue().stream().map(
-                store::getAbstractUnitById).collect(Collectors.toSet()))))));
+            store::getAbstractUnitById).collect(Collectors.toSet())))));
+    this.impossibleCourseModuleAbstractUnits.setData(impossibleCourseModuleAbstractUnits);
+
+    return impossibleCourseModuleAbstractUnits;
```

Comment:
split into a method that computes the data (e.g. in RealReportData) and one step that displays it.